internal/cmd/commands/server/server.go

```diff
@@ -341,6 +343,42 @@ func (c *Command) Run(args []string) int {
 			c.UI.Error(fmt.Errorf("Error connecting to database: %w", err).Error())
 			return 1
 		}
+
+		ctx := context.TODO()
```

Comment:
Use the command's context via `c.Context`

Reply:
Woops, thanks for finding my TODO.  c.Context is never set with the way we initialize the server.Command.  

I've changed that now.  PTAL and let me know if instead of setting our own context the server and dev commands should use base.NewCommand like the others.  I am assuming there was a reason why we didn't want to cancel all context when the Shutdown channel is called (like is set up when NewCommand() is called, but I couldn't figure out why that would be the case.

Reply:
Ah, yeah, you're right. We haven't needed it before (there's only one other place that uses it and that's for something we don't have yet). Ctrl-C is tied to the general shutdown channel -- generally speaking this is because we want to shut down in an orderly fashion instead of having everything shut down/cancel at once. But in this case I think it's a good reason to do more or less the same thing as with NewCommand and populate c.Context, tied to the shutdown channel.

Reply:
Once you do that you can pass that into CreateDevDatabase too.

Reply:
woops.  I had missed a whole handful of context.TODOs.  Thanks again.  I've made the changes and now the `server` and `dev` commands use the base.NewCommand call.